### ISSUE TITLE
[Accessibility] - Plugin window completely inaccessible in MS4, big regression

### ISSUE THREAD

**Issue by hhpmusic** [musescore/MuseScore#17474]
### Issue type

UI bug

### Steps to reproduce

At past in Musescore 3, as long as a plugin has a dialog, we blind can work in it. But in Musescore 4, it doesn't work at all. Just tried, I can enable and disable plugins in the management panel, but all plugins can't be operated using screen reader. When activating a plugin, Musescore will pop up a window where I can do nothing. Our braille music team is trying to do something in the plugin, but it seems we can't go any farther at this time.

### MuseScore Version

4.0.2

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

Windows

**Comment by shoogle** [comment]
It seems it's not possible to focus the dialog at all via the keyboard. It should gain focus automatically as soon as it appears.

Even if you cheat and click inside the dialog with the mouse, keyboard navigation within the dialog is still not possible.

Anything would be better than nothing here, even if it's just normal Qt Quick accessibility within the plugin dialog rather than our new custom accessibility.

**Comment by hhpmusic** [comment]
So is it because of QT or your accessibility implementation? If it's
because of QT, is v6 better than v5?

Haipeng


On 5/10/23, Peter Jonas ***@***.***> wrote:
> It seems it's not possible to focus the dialog at all via the keyboard. It
> should gain focus automatically as soon as it appears.
>
> Even if you cheat and click inside the dialog with the mouse, keyboard
> navigation within the dialog is still not possible.
>
> Anything would be better than nothing here, even if it's just normal Qt
> Quick accessibility within the plugin dialog rather than our new custom
> accessibility.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/musescore/MuseScore/issues/17474#issuecomment-1541200910
> You are receiving this because you authored the thread.
>
> Message ID: ***@***.***>


-- 
Hu Haipeng
Blind musician and braille music transcriber of China
The BrailleOrch and Open Braille Music Projects
http://www.brailleorch.org

**Comment by shoogle** [comment]
@hhpmusic, my guess is it's because we disabled normal Qt QML accessibility in order to create our custom accessibility, but we haven't enabled our custom accessibility on plugins. So I think we need to enable one of those two things on plugins: QML accessibility or our own accessibility. But that's just a guess and could be completely wrong.

We've been [investigating Qt 6](https://github.com/musescore/MuseScore/pull/10108) for a while and determined we're not ready to upgrade yet. There's still too many bugs / missing features at this stage. It wouldn't help with this issue anyway.

**Comment by XiaoMigros** [comment]
Would your own accessibility not work yet? Perhaps unrelated but importing UI theme and components works fine

**Comment by hhpmusic** [comment]
No, we all feel not accessible, Shoogle said already.

**Comment by wizofaus** [comment]
Can put my hand up for this one...

**Comment by igorkorsukov** [comment]
Movements completed...
If we use our UI components, the technical accessibility features should work... But we'll probably need documentation on how to configure them.

**Comment by fcnjd** [comment]
Moren than 2 years have passed since the last comment, and in latest stable MuseScore still we screenreader users cannot interact with plugin windows. Are there any updates so far?
